src/Indice.Features.Identity.UI/Pages/Mfa.cs

```
@@ -155,6 +156,21 @@ private async Task<MfaLoginViewModel> BuildMfaLoginViewModelAsync(string? return
             Error = authenticationMethod == null ? "MFA is enabled but there is no active two factor authentication method configured. Please contact your administrator." : null
         };
     }
+
+    private async Task<TotpResult> SendOtp() {
+        var totpService = TotpServiceFactory.Create<User>();
```

Consider throttling/rate limiting. This essentially means that totp service will not send a totp and will fail if the countdown has not passed.

---

This is already being done by the totp service. When pressing the resend button if you keep hitting the resend you will see the error from the totp service. The error is not propagated OnGetAsync but only OnPostAsync (resend) we should consider maybe showing the error OnGetAsync too.

---

I agree